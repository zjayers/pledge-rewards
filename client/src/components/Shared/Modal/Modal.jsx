import React from 'react';
import ReactDOM from 'react-dom';
import { Fader, InfoCard } from './Modal.style';
import styles from './Modal.module.scss';

import Heading from '../Heading/Heading';
import SubHeading from '../SubHeading/SubHeading';

const Modal = ({ setModalOpen, project }) => {
  const { creator, location } = project;

  const renderModal = () => {
    const toggleModal = (e) => {
      e.stopPropagation();
      setModalOpen(false);
    };

    return (
      <Fader onClick={toggleModal}>
        <InfoCard onClick={(e) => e.stopPropagation()}>
          <p>About the creator</p>
          <Heading heavy>{creator}</Heading>
          <SubHeading>{location}</SubHeading>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Adipisci
            consequatur dicta, eveniet exercitationem laborum minus mollitia
            neque odio perferendis, placeat quod rem ut voluptates. Consequuntur
            est in iure minima perspiciatis? Lorem ipsum dolor sit amet,
            consectetur adipisicing elit. Ad cumque doloremque maxime, nihil non
            omnis sapiente. Commodi inventore molestias ratione reiciendis
            sequi. Asperiores beatae eos id nam quam quidem vero! Lorem ipsum
            dolor sit amet, consectetur adipisicing elit. Ad corporis deleniti,
            distinctio esse, labore modi odit officiis pariatur provident qui
            rerum veniam veritatis! Ad doloremque fuga harum suscipit vitae.
            Modi! Lorem ipsum dolor sit amet, consectetur adipisicing elit.
            Corporis cupiditate temporibus voluptate! Aliquid deserunt dicta
            expedita hic libero nihil qui sed. Cumque debitis dolorem earum enim
            ipsa modi tempora ut!
          </p>
          <div className={styles.flex}>
            <div className={styles.flexColumn}>
              <ul>
                <li>
                  <i className='fas fa-check' style={{ color: '#009E74' }} />
                </li>
                <li>
                  <i className='fas fa-lock' />
                </li>
                <li>
                  <i className='fab fa-facebook' style={{ color: '#39579A' }} />
                </li>
                <li>
                  <i
                    className='fab fa-kickstarter-k'
                    style={{ color: '#057362' }}
                  />
                </li>
              </ul>
            </div>
            <div className={styles.flexColumn}>
              <ul>
                <li>{creator}</li>
                <li>Last login</li>
                <li>Not connected</li>
                <li style={{ color: '#27A47E' }}>1 created Â· 0 backed</li>
              </ul>
            </div>
          </div>
        </InfoCard>
        <span>
          {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */}
          <u onClick={toggleModal}>Close</u>
        </span>
      </Fader>
    );
  };

  return ReactDOM.createPortal(
    renderModal(),
    document.getElementById('modal-root')
  );
};

export default Modal;
