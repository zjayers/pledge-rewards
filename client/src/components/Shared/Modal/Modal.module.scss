.flex {
  display: flex;
}

.flexColumn {
  display: flex;
  flex-direction: column;

  & ul {
    list-style: none;
  }

  & li {
    font-size: 12px;
    margin: 12px 10px 12px 0;
  }
}
