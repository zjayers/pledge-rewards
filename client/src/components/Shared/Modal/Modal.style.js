import styled from 'styled-components';

export const Fader = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  background: rgba(255, 255, 255, 0.9);
  display: flex;
  flex-direction: column;

  & span {
    display: inline;
    font-size: 12px;
    position: absolute;
    left: 0;
    right: 0;
    bottom: 12vh;
    text-align: center;
  }

  & span u {
    cursor: pointer;
    color: #656969;
    &:hover {
      color: #28282b;
    }
  }
`;

export const InfoCard = styled.div`
  width: 40vw;
  max-height: 70vh;
  box-shadow: 0 0 4px 0 rgba(168, 167, 164, 0.4);
  padding: 2.4rem;
  overflow-y: auto;
  color: #282828;
  z-index: 999;
  left: 0;
  right: 0;
  top: 0;
  margin: auto;

  & p:first-of-type {
    font-size: 18px;
    text-align: center;
    margin: 0 auto 60px;
  }

  & h2 {
    font-size: 2.4rem;
  }

  & p:last-of-type {
    margin-top: 10px;
    font-size: 14px;
    line-height: 1.4;
    margin-bottom: 30px;
  }
`;
