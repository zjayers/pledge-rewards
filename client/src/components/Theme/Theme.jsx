import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  colors: {
    borderGrey: '#9B9E9E',
    highlightGreen: '#009E74',
    buttonGreen: '#009E74',
    gradientBlue: '#DBE7FE',
    gradientOrange: '#FFF2EC',
    text: '#282828',
    flavorText: '#888B8B',
    linkText: '#498F80',
  },
  fonts: ['Muli', 'sans-serif'],
  fontSizes: {
    small: '1rem',
    medium: '1.8rem',
    large: '3rem',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
