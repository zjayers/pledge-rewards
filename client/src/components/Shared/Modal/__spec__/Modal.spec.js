import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Modal from '../Modal';
import Theme from '../../../Theme/Theme';
import { Fader, InfoCard } from '../Modal.style';

describe('Modal Component', () => {
  let component;

  const project = {
    creator: 'Bob',
    location: 'Home',
  };

  const mockModalFn = jest.fn();

  // add a div with #modal-root id to the global body
  const modalRoot = global.document.createElement('div');
  modalRoot.setAttribute('id', 'modal-root');
  const body = global.document.querySelector('body');
  body.appendChild(modalRoot);

  beforeEach(() => {
    component = mount(
      <Theme>
        <Modal setModalOpen={mockModalFn} project={project} />
      </Theme>
    );
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should match the test snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should hide the modal when the close button is clicked', () => {
    const fader = component.find(Fader);
    fader.simulate('click', () => {});
    expect(mockModalFn).toHaveBeenCalled();
  });

  it('should not close the modal when the info card is clicked', () => {
    const fader = component.find(InfoCard);
    fader.simulate('click', () => {});
    expect(mockModalFn).toHaveBeenCalledTimes(0);
  });
});
