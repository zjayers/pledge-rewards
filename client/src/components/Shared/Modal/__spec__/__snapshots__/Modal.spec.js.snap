// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should match the test snapshot 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <Modal
      project={
        Object {
          "creator": "Bob",
          "location": "Home",
        }
      }
      setModalOpen={[MockFunction]}
    >
      <Portal
        containerInfo={
          <div
            id="modal-root"
          >
            <div
              class="Modalstyle__Fader-j211o6-0 dTHFpQ"
            >
              <div
                class="Modalstyle__InfoCard-j211o6-1 uIpWh"
              >
                <p>
                  About the creator
                </p>
                <h2
                  class="Headingstyle___default-dlpxpq-0 iWDnYS"
                >
                  Bob
                </h2>
                <p
                  class="SubHeadingstyle___default-qgikex-0 eiBTbF"
                >
                  Home
                </p>
                <p>
                  Lorem ipsum dolor sit amet, consectetur adipisicing elit. Adipisci consequatur dicta, eveniet exercitationem laborum minus mollitia neque odio perferendis, placeat quod rem ut voluptates. Consequuntur est in iure minima perspiciatis? Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad cumque doloremque maxime, nihil non omnis sapiente. Commodi inventore molestias ratione reiciendis sequi. Asperiores beatae eos id nam quam quidem vero! Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad corporis deleniti, distinctio esse, labore modi odit officiis pariatur provident qui rerum veniam veritatis! Ad doloremque fuga harum suscipit vitae. Modi! Lorem ipsum dolor sit amet, consectetur adipisicing elit. Corporis cupiditate temporibus voluptate! Aliquid deserunt dicta expedita hic libero nihil qui sed. Cumque debitis dolorem earum enim ipsa modi tempora ut!
                </p>
                <div
                  class="flex"
                >
                  <div
                    class="flexColumn"
                  >
                    <ul>
                      <li>
                        <i
                          class="fas fa-check"
                          style="color: rgb(0, 158, 116);"
                        />
                      </li>
                      <li>
                        <i
                          class="fas fa-lock"
                        />
                      </li>
                      <li>
                        <i
                          class="fab fa-facebook"
                          style="color: rgb(57, 87, 154);"
                        />
                      </li>
                      <li>
                        <i
                          class="fab fa-kickstarter-k"
                          style="color: rgb(5, 115, 98);"
                        />
                      </li>
                    </ul>
                  </div>
                  <div
                    class="flexColumn"
                  >
                    <ul>
                      <li>
                        Bob
                      </li>
                      <li>
                        Last login
                      </li>
                      <li>
                        Not connected
                      </li>
                      <li
                        style="color: rgb(39, 164, 126);"
                      >
                        1 created · 0 backed
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
              <span>
                <u>
                  Close
                </u>
              </span>
            </div>
          </div>
        }
      >
        <Modalstyle__Fader
          onClick={[Function]}
        >
          <div
            className="Modalstyle__Fader-j211o6-0 dTHFpQ"
            onClick={[Function]}
          >
            <Modalstyle__InfoCard
              onClick={[Function]}
            >
              <div
                className="Modalstyle__InfoCard-j211o6-1 uIpWh"
                onClick={[Function]}
              >
                <p>
                  About the creator
                </p>
                <_default
                  heavy={true}
                >
                  <Headingstyle___default
                    heavy={true}
                  >
                    <h2
                      className="Headingstyle___default-dlpxpq-0 iWDnYS"
                    >
                      Bob
                    </h2>
                  </Headingstyle___default>
                </_default>
                <_default>
                  <SubHeadingstyle___default>
                    <p
                      className="SubHeadingstyle___default-qgikex-0 eiBTbF"
                    >
                      Home
                    </p>
                  </SubHeadingstyle___default>
                </_default>
                <p>
                  Lorem ipsum dolor sit amet, consectetur adipisicing elit. Adipisci consequatur dicta, eveniet exercitationem laborum minus mollitia neque odio perferendis, placeat quod rem ut voluptates. Consequuntur est in iure minima perspiciatis? Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad cumque doloremque maxime, nihil non omnis sapiente. Commodi inventore molestias ratione reiciendis sequi. Asperiores beatae eos id nam quam quidem vero! Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad corporis deleniti, distinctio esse, labore modi odit officiis pariatur provident qui rerum veniam veritatis! Ad doloremque fuga harum suscipit vitae. Modi! Lorem ipsum dolor sit amet, consectetur adipisicing elit. Corporis cupiditate temporibus voluptate! Aliquid deserunt dicta expedita hic libero nihil qui sed. Cumque debitis dolorem earum enim ipsa modi tempora ut!
                </p>
                <div
                  className="flex"
                >
                  <div
                    className="flexColumn"
                  >
                    <ul>
                      <li>
                        <i
                          className="fas fa-check"
                          style={
                            Object {
                              "color": "#009E74",
                            }
                          }
                        />
                      </li>
                      <li>
                        <i
                          className="fas fa-lock"
                        />
                      </li>
                      <li>
                        <i
                          className="fab fa-facebook"
                          style={
                            Object {
                              "color": "#39579A",
                            }
                          }
                        />
                      </li>
                      <li>
                        <i
                          className="fab fa-kickstarter-k"
                          style={
                            Object {
                              "color": "#057362",
                            }
                          }
                        />
                      </li>
                    </ul>
                  </div>
                  <div
                    className="flexColumn"
                  >
                    <ul>
                      <li>
                        Bob
                      </li>
                      <li>
                        Last login
                      </li>
                      <li>
                        Not connected
                      </li>
                      <li
                        style={
                          Object {
                            "color": "#27A47E",
                          }
                        }
                      >
                        1 created · 0 backed
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </Modalstyle__InfoCard>
            <span>
              <u
                onClick={[Function]}
              >
                Close
              </u>
            </span>
          </div>
        </Modalstyle__Fader>
      </Portal>
    </Modal>
  </ThemeProvider>
</Theme>
`;
